wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] Retry view

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2740

## To test:
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

📓 If you've played around with this feature recently then it would be best to start from a fresh build to make sure you aren't seeing cached data
 - Layouts are cached per site. So to help testing if you fetch the data on site A, you can put the device in airplane mode again and test on site B.

### No Network Connection Displays
1. Make sure your device is not connected to the internet
1. Navigate to Modal Layout Picker
    - **Expect** to see the no network connection appear

<kbd><img width="300" alt="Screen Shot 2020-10-20 at 4 01 02 PM" src="https://user-images.githubusercontent.com/3384451/96639452-bcdcc200-12ef-11eb-9bb1-ee6fac4082f4.png"></kbd>

### Retry works
1. Make sure your device is not connected to the internet
1. Navigate to Modal Layout Picker
    - **Expect** to see the no network connection appear
1. Re-establish a network connection
1. Select "Retry"
    - **Expect** to see the data load

### Quick Sanity
1. Make sure your device is not connected to the internet
1. Navigate to Modal Layout Picker for a site where you have opened the picker before
    - **Expect** to see the data load

### PR submission checklist:
